GHCN.py

```diff
@@ -0,0 +1,1044 @@
+# coding: utf-8
+'''
+Created on 2014-08-20
+
+A module to load station daily data from Environment Canada from ASCII files and convert them to monthly 
+NetCDF datasets (with extremes); the module also provides a wrapper to load the NetCDF datasets. 
+
+@author: Andre R. Erler, GPL v3
+'''
+
+# external imports
+import numpy as np
+from copy import deepcopy
+import codecs, calendar, functools
+from warnings import warn
+from collections import OrderedDict
+# internal imports
+from datasets.CRU import loadCRU_StnTS
+from datasets.common import days_per_month, data_root, selectElements, translateVarNames
+from datasets.common import CRU_vars, stn_params, nullNaN
+from geodata.misc import ParseError, DateError, VariableError, ArgumentError, DatasetError, AxisError
+from geodata.misc import RecordClass, StrictRecordClass, isNumber, isInt 
+from geodata.base import Axis, Variable, Dataset
+from utils.nctools import writeNetCDF
+from geodata.netcdf import DatasetNetCDF
+from geodata.gdal import NamedShape, ShapeInfo
+# import derived variables from the WRF Tools package wrfavg
+import imp, os
+# read code root folder from environment variable
+code_root = os.getenv('CODE_ROOT')
+if not code_root : raise ArgumentError, 'No CODE_ROOT environment variable set!'
+if not os.path.exists(code_root): raise ImportError, "The code root '{:s}' directory set in the CODE_ROOT environment variable does not exist!".format(code_root)
+# import module from WRF Tools explicitly to avoid name collision
+dv = imp.load_source('derived_variables', os.getenv('CODE_ROOT')+'/WRF-Tools/Python/wrfavg/derived_variables.py') # need explicit absolute import due to name collision
+#dv = importlib.import_module('wrfavg.derived_variables') # need explicit absolute import due to name collision
+#import wrfavg.derived_variables as dv
+from utils.constants import precip_thresholds
+#from wrfavg.derived_variables import precip_thresholds
+
+## GHCN (Environment Canada) Meta-data
+
+dataset_name = 'GHCN'
+root_folder = '{:s}/{:s}/'.format(data_root,dataset_name) # long-term mean folder
+orig_ts_file = '{1:s}.dly' # filename pattern: variable name and station ID
+tsfile = 'ghcn{0:s}_monthly.nc' # filename pattern: station type
+#tsfile_prov = 'ghcn{0:s}_{1:s}_monthly.nc' # filename pattern with province: station type, province  
+avgfile = 'ghcn{0:s}_clim{1:s}.nc' # filename pattern: station type and ('_'+period)
+avgfolder = root_folder + 'ghcnavg/'  # folder for user data
+
+
+# variable attributes and name
+varatts = dict(T          = dict(name='T', units='C', atts=dict(long_name='Average Temperature')), # average temperature
+               Tmin       = dict(name='Tmin', units='C', atts=dict(long_name='Minimum Temperature')), # minimum temperature

```

Comment:
I've removed unused definitions from the EC module as well; you can have a look there.